Added the function stringToLetterArray to convert string to array

JoodyKonno opened the pull request:
The function stringToLetterArray takes a string and returns an array of its letters. The output of this function can be used as an input to itemCounts function.

Example: 
```javascript
const { stringToLetterArray } = require('./textalyze');
stringToLetterArray('aaabcc');
// returns ['a', 'a', 'a', 'b', 'c', 'c']
```

Commit:
Added stringToLetterArray function

The input of the itemCounts is more likely to be a string than an array of letters. It can be solved with a .split() function but it`s repetitive.

To avoid code duplication, a helper function stringToLetterArray was added to encapsulate the logic

Commit:
Export the stringToLetterArray as a module function

The itemCounts and stringToLetterArray are more likely to run together, so exported the stringToLetterArray function, to be used by the application

jfarmer commented on textalyze.js:
```diff
@@ -16,18 +16,20 @@ const itemCounts = array => array.reduce((counts, item) => !counts.has(item)
   : counts.set(item, counts.get(item) + 1)
   , new Map());
 
+const stringToLetterArray = str => str.split('');
```
Great.  For practice, write a few tests for this function in the same style as the tests in `__tests__`.

Commit:
Added tests for the stringToLetterArray function

Added two tests for the stringToLetterArray function.

- test whether it return the correct array of letter for the given string
- test wheter it accepts a non-string input. If a non-string value is passed it returns an empty array
